site/src/book/filtering-expression.md

```diff
@@ -0,0 +1,47 @@
+# Filtering expression
+
+* **Introduced in:** cargo-nextest 0.9.13
+* **Environment variable**: `NEXTEST_EXPERIMENTAL_EXPR_FILTER=1`
+* **Tracking issue**: []
+
+Tests to run can be filtered using filter expressions.
+
+## Filtering expression DSL
+
+A filtering expression define a set of tests, any test in the set will be run.
+
+Basic sets:
+- `all()`: include everything
+- `test(name-matcher)`: include all tests matching `name-matcher`
+- `package(name-matcher)`: include all tests in packages matching `name-matcher`
+- `deps(name-matcher)`: include all tests in packages depended by packages matching `name-matcher` and include all tests in packages matching `name-matcher`
+- `rdeps(name-matcher)`: include all tests in packages depending on packages matching `name-matcher` and include all tests in packages matching `name-matcher`
+- `none()`: include nothing
+
+Name matcher:
+- `text`: match anything containing `text`
+- `'unicode_text'`: match anything containing `unicode_text`
+- `=text`: match anything equal to `text`
+- `='unicode_text'`: match anything equal to `unicode_text`
+- `/reg/`: match anything matching the regex `reg`
+
+Unicode string:
+- can contains escaped quote: `\'`
+- can contains unicode sequence: `\u{xxx}` (where `xxx` is an 1 to 6 digits hexadecimal number)
+
+Operations:
+- `set_1 & set_2` , `set_1 and set_2`: include everything in both `set_1` and `set_2`
+- `set_1 | set_2`, `set_1 or set_2`, `set_1 + set_2`: include everything in either `set_1` or `set_2`
+- `not set`: include everything not included in `set`
+- `set_1 - set_2`: equivalent to `set_1 and not set_2`
```

Comment:
What is the operator precedence here -- are there tests for that? Could you document this?

Reply:
Doc added.

There is a test for it `nextest-filtering parsing::test_parse_epxr_precedence`